### ISSUE TITLE
About roll angle rectify after image flipping

### ISSUE THREAD

**Issue by sunnyykk** [facebookresearch/OrienterNet#50]
Hello,

I noticed that in [line 159](https://github.com/facebookresearch/OrienterNet/blob/c222d08c9c0a084d79a415cc5c43e71173d2ecc4/maploc/data/dataset.py#L159), you employed random flipping to augment the query and map images, and later, in [line 164](https://github.com/facebookresearch/OrienterNet/blob/c222d08c9c0a084d79a415cc5c43e71173d2ecc4/maploc/data/dataset.py#L164), the query image are rectified by setting the roll and pitch angles to zero. However, I'm wondering if a negative sign should be applied to the [roll](https://github.com/facebookresearch/OrienterNet/blob/c222d08c9c0a084d79a415cc5c43e71173d2ecc4/maploc/data/dataset.py#L214) angle during the rectification process after the images have been flipped. 

I'm quite new to this area, thank you for your attention to this detail!

**Comment by sunnyykk** [comment]
To further elaborate on the issue mentioned above, some visualizations are provided:
- The first image below is a training image after `random_flip` function, which includes mirrored Chinese characters;
- The following image is after gravity correction using the `process_image` function; note that the lateral roll angle has become larger;
- The third image is the result after I took the negative value of the roll angle and reprocess it;

![image](https://github.com/facebookresearch/OrienterNet/assets/34845957/271e79df-e38b-4582-bd75-831e68b03643)

![image](https://github.com/facebookresearch/OrienterNet/assets/34845957/bb28810c-8f97-41e4-9bc1-6add51f3c32b)

![image](https://github.com/facebookresearch/OrienterNet/assets/34845957/f57059d8-1489-42de-82ae-d79031bc0abb)

Is it possible that there is a bug in the gravity correction process during training that might potential affect half of the augmented training samples?

**Comment by sarlinpe** [comment]
You are right, thank you investigating and reporting this! I likely missed it on visual inspection because most images of our dataset have zero roll. I still think that fixing it will improve the model. Could you submit a PR for it? I'll retrain the model once I find the time.

**Comment by sunnyykk** [comment]
thanks! I will submit it~
